### PR TITLE
SAK-41502 Add column GB_GRADING_EVENT_T.IS_EXCLUDED to 20.x scripts

### PR DESCRIPTION
**Pull request by ern**
https://jira.sakaiproject.org/browse/SAK-41502

### CODE REVIEW

**Review by ottenhoff — APPROVED** [review]

**Review by bjones86 — APPROVED** [review]
Thanks for the Oracle script!

File: docs/conversion/sakai_20_2-20_3_mysql_conversion.sql [hunk 1]
```diff
@@ -19,3 +19,7 @@ ALTER TABLE lti_tools RENAME COLUMN allowsettings TO allowsettings_ext;
 -- UPDATE lti_tools SET allowsettings_ext=allowsettings;
 -- ALTER TABLE lti_tools DROP COLUMN allowsettings;
 -- END SAK-44810
+
+-- SAK-41502
+ALTER TABLE GB_GRADING_EVENT_T ADD IS_EXCLUDED INT DEFAULT null NULL;
```
Comment (bjones86) [inline docs/conversion/sakai_20_2-20_3_mysql_conversion.sql:24]: The original PR is slightly different:

`ALTER TABLE GB_GRADING_EVENT_T ADD IS_EXCLUDED INTEGER NOT NULL DEFAULT 0;`

File: docs/conversion/sakai_20_2-20_3_oracle_conversion.sql [hunk 2]
```diff
@@ -19,3 +19,7 @@ ALTER TABLE lti_tools RENAME COLUMN allowsettings TO allowsettings_ext;
 -- UPDATE lti_tools SET allowsettings_ext=allowsettings;
 -- ALTER TABLE lti_tools DROP COLUMN allowsettings;
 -- END SAK-44810
+
+-- SAK-41502
+ALTER TABLE GB_GRADING_EVENT_T ADD IS_EXCLUDED NUMBER(10, 0);
```
Comment (bjones86) [inline docs/conversion/sakai_20_2-20_3_oracle_conversion.sql:24]: The original PR is slightly different:

`ALTER TABLE GB_GRADING_EVENT_T ADD IS_EXCLUDED NUMBER(1,0) DEFAULT 0 NOT NULL;`

**Review by bjones86 — CHANGES_REQUESTED** [review]
On second thought, I have some questions about this:

1) Why do we need this PR? These column additions are already present in the 20 conversion script, introduced here: #90 

2) The data types different from the original implementation

3) There is no default values here, but the original provides them

File: docs/conversion/sakai_20_2-20_3_oracle_conversion.sql (hunk 2 above)
Comment (ern) [reply docs/conversion/sakai_20_2-20_3_oracle_conversion.sql:24]: SAK-41502 was never merged to 20.x until yesterday, which adds the column, I would say the 20.0 script is wrong.
And I think it's been modified from the original.

Comment (ern) [reply docs/conversion/sakai_20_2-20_3_oracle_conversion.sql:24]: I've updated to remove the 20.0 conversion

File: docs/conversion/sakai_20_2-20_3_mysql_conversion.sql (hunk 1 above)
Comment (ern) [reply docs/conversion/sakai_20_2-20_3_mysql_conversion.sql:24]: I was going off what that code says according to the hbm.

File: docs/conversion/sakai_20_2-20_3_oracle_conversion.sql (hunk 2 above)
Comment (bjones86) [reply docs/conversion/sakai_20_2-20_3_oracle_conversion.sql:24]: I agree that it should only be in 1 script, not 2. However, I'm not convinced that the old ones were "wrong". Can you elaborate? I specifically asked for this column to be not-nullable and have a proper default in the sakaiproject/sakai PR almost a year ago.

File: docs/conversion/sakai_20_2-20_3_mysql_conversion.sql (hunk 1 above)
Comment (ern) [reply docs/conversion/sakai_20_2-20_3_mysql_conversion.sql:24]: If it should be non null and have a default then the code/hbm are incorrect.

Comment (bjones86) [reply docs/conversion/sakai_20_2-20_3_mysql_conversion.sql:24]: I asked for it to be non-null and have a default value [here](https://github.com/sakaiproject/sakai/pull/8364#discussion_r454433678) and [here](https://github.com/sakaiproject/sakai-reference/pull/90#pullrequestreview-448192191).

File: docs/conversion/sakai_20_2-20_3_oracle_conversion.sql (hunk 2 above)
Comment (bjones86) [reply docs/conversion/sakai_20_2-20_3_oracle_conversion.sql:24]: Do we really need this column to be able to store numbers up to 10 billion? That seems excessive.

**Comment by plukasew** [comment]
This comment represents an enum value that indicates if there was a status change to the grade. The possible values are 0 (no change), 1 (grade was excluded), 2 (grade was included). I think the obvious default value here is 0 and there is no reason to allow this to be nullable. The name of the column "is_excluded" is also problematic because it implies a boolean when it is not. Since this is a very small integer value, it can be stored in NUMBER(1,0) on Oracle.

**Comment by ern** [comment]
Thanks Paul and Bryan however this is not a change authored by me I suggest we got back to @victorGomollon and ask him to correct it. Can you comment in the existing JIRA a open a new with the changes your proposing.

**Comment by bjones86** [comment]
Okay sure. Should we close this for now then? Or can we make the appropriate changes here and open a new ticket to make the corresponding changes in code/hbm?

**Comment by ern** [comment]
We can leave this open for right now so we don't lose sight of it but we need a new JIRA and code change for the updates your proposing, I don't mind if we just end up updating this once the code change is in.

**Comment by bjones86** [comment]
Okay sounds good. I've created https://jira.sakaiproject.org/browse/SAK-45258 and linked it to SAK-41502. I'm also going to slap the "in progress" label here, just so we don't accidentally merge without making the required changes.

**Comment by ern** [comment]
This has been merged as it's needed for 20.3 scripts

**Pull request merged**